components/email-form.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { Link } from 'react-router';
+import validateEmailForm from './validate-email-form';
+
+export default class EmailForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const { success, errors, value } = props;
+    this.state = {
+      success,
+      errors,
+      value: value || ''
+    };
+    this.formSubmit = this.formSubmit.bind(this);
+    this.inputChange = this.inputChange.bind(this);
+    this.resetForm = this.resetForm.bind(this);
+  }
+
+  formSubmittedSuccessfully() {
+    return this.state.success === true;
+  }
+
+  inputChange(e) {
+    this.setState({
+      value: e.target.value
+    });
+  }
+
+  resetForm(e) {
+    e.preventDefault();
+    this.setState({
+      success: undefined,
+      errors: [],
+      value: ''
+    });
+  }
+
+  formSubmit(e) {
+    e.preventDefault();
+    const emailValue = this.state.value;
+    // note how this is the same validator we ran on the server!
+    const result = validateEmailForm(emailValue);
+    this.setState({
```

Comment:
In a real app you'd probably do something here to store this piece of data - most likely hitting an API endpoint. The API endpoint wouldn't do any validation though, just store into the database, because the validation has been done on the client.

Alternatively you could just do the validation all on the server and return all the validation input, but why do that when we can have it all on the client? :)